src/create_disk.sh

```diff
@@ -79,18 +88,56 @@ sgdisk -Z $disk \
 	-n 4:0:0     -c 4:root       -t 4:4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709
 sgdisk -p "$disk"
 
-udevadm trigger
-udevadm settle
+udevtrig
+
+root_dev="${disk}4"
+if [ -n "${luks_rootfs}"  ]; then
+    root_dev=/dev/mapper/crypt_root
+    sgdisk -c 4:luk_root "${disk}"
+
+    touch tmp.key
+    # Create the LUKS partition using the null_cipher and a sentinal
+    # UUID similiar to the one used by coreos-gpt-setup. This is used
+    # by ignition-dracut-reecrypt.
+    cryptsetup luksFormat \
+        -q \
+        --label="crypt_rootfs" \
+        --cipher=cipher_null \
+        --key-file=tmp.key \
+        --uuid='00000000-0000-4000-a000-000000000002' \
+        "${disk}4"
+
+    # 'echo ""' acts as a test that you can use an empty
+    # password. You can actually use _any_ string.
+    echo "" | cryptsetup luksOpen \
+        --allow-discards \
+        "${disk}4" crypt_root \
+        --key-file=-
+
+    udevtrig
+
+    cryptsetup token import \
+        "${disk}4" \
+        --token-id 9 \
+        --key-slot=0 \
+        <<<'{"type": "coreos", "keyslots": ["0"], "key": "", "ostree_ref": "'${ref}'"}'
```

Comment:
Hm, is the `ostree_ref` to distinguish the target OS?  `${os_name}` is probably a bit nicer here, since RHCOS doesn't use a ref really.